tool/tbot/config/destination_directory.go

```diff
@@ -58,23 +61,142 @@ func (dd *DestinationDirectory) CheckAndSetDefaults() error {
 		return trace.BadParameter("destination path must not be empty")
 	}
 
+	secureSupported, err := botfs.HasSecureWriteSupport()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	aclsSupported, err := botfs.HasACLSupport()
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	switch dd.Symlinks {
```

Comment:
Nit: I'd consider implementing CheckAndSetDefaults on `SymlinksMode` and `ACLMode` that do this.